src/ansys/fluent/core/launcher/standalone_launcher.py

```diff
@@ -0,0 +1,288 @@
+        case_data_file_name : str, optional
+            If provided, the case and data files at ``case_data_file_name`` are read into the Fluent session.
+        lightweight_mode : bool, optional
+            Whether to run in lightweight mode. In lightweight mode, the lightweight settings are read into the
+            current Fluent solver session. The mesh is read into a background Fluent solver session which will
+            replace the current Fluent solver session once the mesh read is complete and the lightweight settings
+            made by the user in the current Fluent solver session have been applied in the background Fluent
+            solver session. This is all orchestrated by PyFluent and requires no special usage.
+            This parameter is used only when ``case_file_name`` is provided. The default is ``False``.
+        mode : str, optional
+            Launch mode of Fluent to point to a specific session type.
+            The default value is ``None``. Options are ``"meshing"``,
+            ``"pure-meshing"`` and ``"solver"``.
+        py : bool, optional
+            If True, Fluent will run in Python mode. Default is None.
+        gpu : bool, optional
+            If True, Fluent will start with GPU Solver.
+        cwd : str, Optional
+            Working directory for the Fluent client.
+        topy : bool or str, optional
+            A boolean flag to write the equivalent Python journal(s) from the journal(s) passed.
+            Can optionally take the file name of the new python journal file.
+        start_watchdog : bool, optional
+            When ``cleanup_on_exit`` is True, ``start_watchdog`` defaults to True,
+            which means an independent watchdog process is run to ensure
+            that any local GUI-less Fluent sessions started by PyFluent are properly closed (or killed if frozen)
+            when the current Python process ends.
+
+        Returns
+        -------
+        :obj:`~typing.Union` [:class:`Meshing<ansys.fluent.core.session_meshing.Meshing>`, \
+        :class:`~ansys.fluent.core.session_pure_meshing.PureMeshing`, \
+        :class:`~ansys.fluent.core.session_solver.Solver`, \
+        :class:`~ansys.fluent.core.session_solver_icing.SolverIcing`, dict]
+            Session object or configuration dictionary if ``dry_run = True``.
+
+        Raises
+        ------
+        UnexpectedKeywordArgument
+            If an unexpected keyword argument is provided.
+        DockerContainerLaunchNotSupported
+            If a Fluent Docker container launch is not supported.
+
+        Notes
+        -----
+        Job scheduler environments such as SLURM, LSF, PBS, etc. allocates resources / compute nodes.
+        The allocated machines and core counts are queried from the scheduler environment and
+        passed to Fluent.
+        """
+        _process_kwargs(kwargs)
+        del kwargs
+        fluent_launch_mode = _get_fluent_launch_mode(
```

Comment:
@hpohekar We're depending heavily on this function throughout these concrete launchers. But the mode is already decided when we choose to instantiate the launcher - that's why we have the factory.

Reply:
@seanpearsonuk Got it. It should be used only in `launch_fluent()`. We will update it. Thank you.

Reply:
Done. Thank you.